File: exercism_data/src/406.rb
class Phrase
  def initialize(phrase)
    @words = phrase.downcase.split(/\W+/)
  end

  def word_count
    @words.uniq.each_with_object({}) do |word, counted_words|
      counted_words[word] = @words.count(word)
    end
  end
end

File: assignments/python/wc/src/406.py
import string
def word_count(phrase):
  counts = {}
  phrase = phrase.translate(None, string.punctuation).lower()
  words = [token for token in phrase.split(' ') if token.strip() != '']
  for word in words:
    if word in counts:
      counts[word] += 1
    else:
      counts[word] = 1
  return counts
